crates/oxc_linter/src/rules/eslint/prefer_numeric_literals.rs

```diff
@@ -0,0 +1,301 @@
+        "Number.parseInt('17', 8)+5",
+        "parseInt('A', 16)+5",
+        "parseInt('11', 2)in foo",
+        "Number.parseInt('17', 8)in foo",
+        "parseInt('A', 16)in foo",
+        "parseInt('11', 2) in foo",
+        "Number.parseInt('17', 8)/**/in foo",
+        "(parseInt('A', 16))in foo",
+        "/* comment */Number.parseInt('11', 2);",
+        "Number/**/.parseInt('11', 2);",
+        "Number//
+			.parseInt('11', 2);",
+        "Number./**/parseInt('11', 2);",
+        "Number.parseInt(/**/'11', 2);",
+        "Number.parseInt('11', /**/2);",
+        "Number.parseInt('11', 2)/* comment */;",
+        "parseInt/**/('11', 2);",
+        "parseInt(//
+			'11', 2);",
+        "parseInt('11'/**/, 2);",
+        "parseInt(`11`/**/, 2);",
+        "parseInt('11', 2 /**/);",
+        "parseInt('11', 2)//comment
+			;",
+        r#"parseInt?.("1F7", 16) === 255;"#,
+        r#"Number?.parseInt("1F7", 16) === 255;"#,
+        r#"Number?.parseInt?.("1F7", 16) === 255;"#,
+        r#"(Number?.parseInt)("1F7", 16) === 255;"#,
+        r#"(Number?.parseInt)?.("1F7", 16) === 255;"#,
+        "parseInt('1_0', 2);",
+        "Number.parseInt('5_000', 8);",
+        "parseInt('0_1', 16);",
+        "Number.parseInt('0_0', 16);",
+    ];
+
+    let fix = vec![
+        (r#"parseInt("111110111", 2) === 503;"#, "0b111110111 === 503;", None),
+        (r#"parseInt("767", 8) === 503;"#, "0o767 === 503;", None),
+        (r#"parseInt("1F7", 16) === 255;"#, "0x1F7 === 255;", None),
+        (r#"Number.parseInt("111110111", 2) === 503;"#, "0b111110111 === 503;", None),
+        (r#"Number.parseInt("767", 8) === 503;"#, "0o767 === 503;", None),
+        (r#"Number.parseInt("1F7", 16) === 255;"#, "0x1F7 === 255;", None),
+        ("parseInt(`111110111`, 2) === 503;", "0b111110111 === 503;", None),
+        ("parseInt(`767`, 8) === 503;", "0o767 === 503;", None),
+        ("parseInt(`1F7`, 16) === 255;", "0x1F7 === 255;", None),
+        ("parseInt('11', 2)", "0b11", None),
+        ("Number.parseInt('67', 8)", "0o67", None),
+        ("5+parseInt('A', 16)", "5+0xA", None),
+        // ("function *f(){ yield(Number).parseInt('11', 2) }", "function *f(){ yield 0b11 }", None),
+        // ("function *f(){ yield(Number.parseInt)('67', 8) }", "function *f(){ yield 0o67 }", None),
+        // ("function *f(){ yield(parseInt)('A', 16) }", "function *f(){ yield 0xA }", None),
+        // ("function *f(){ yield Number.parseInt('11', 2) }", "function *f(){ yield 0b11 }", None),
+        // (
+        //     "function *f(){ yield/**/Number.parseInt('67', 8) }",
+        //     "function *f(){ yield/**/0o67 }",
+        //     None,
+        // ),
+        // ("function *f(){ yield(parseInt('A', 16)) }", "function *f(){ yield(0xA) }", None),
+        ("parseInt('11', 2)+5", "0b11+5", None),
+        ("Number.parseInt('17', 8)+5", "0o17+5", None),
+        ("parseInt('A', 16)+5", "0xA+5", None),
+        // ("parseInt('11', 2)in foo", "0b11 in foo", None),
```

Comment:
Why don't these cases work?

Reply:
These cases don't work as the fixer will create invalid Javascript, for example:
```
  left: "0b11 in foo"
 right: "0b11in foo"
```

Reply:
Decided to drop the code fixer for now, I can work on that as an improvement.